library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/dynamic/DynamicMetaregistryImpl.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.impl.dynamic;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableMap;
+import com.mojang.serialization.Codec;
+
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.RegistryLoader;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.registry.mixin.DynamicRegistrySyncAccessor;
+
+public class DynamicMetaregistryImpl {
```

Comment:
```suggestion
@ApiStatus.Internal
public class DynamicMetaRegistryImpl {
```

Will also need the import in case you try to add my suggestions directly in the GitHub UI.

Reply:
You forgot to rename the API class too :P

Reply:
I missed that oops, anyway this needs IDE intervention anyways.